api/bucketsd/client/client_test.go

```diff
@@ -712,6 +712,111 @@ func setPath(t *testing.T, private bool) {
+	})
+
+	t.Run("private", func(t *testing.T) {
+		move(t, ctx, client, true)
+	})
+}
+
+func move(t *testing.T, ctx context.Context, client *c.Client, private bool) {
+	buck, err := client.Create(ctx, c.WithPrivate(private))
+	require.NoError(t, err)
+
+	q, err := client.PushPaths(ctx, buck.Root.Key)
+	require.NoError(t, err)
+	err = q.AddFile("root.jpg", "testdata/file1.jpg")
+	require.NoError(t, err)
+	err = q.AddFile("a/b/c/file1.jpg", "testdata/file1.jpg")
+	require.NoError(t, err)
+	err = q.AddFile("a/b/c/file2.jpg", "testdata/file2.jpg")
+	require.NoError(t, err)
+	for q.Next() {
+		require.NoError(t, q.Err())
+	}
+	q.Close()
+
+	// move a dir into a new path  => a/c
+	_, err = client.MovePath(ctx, buck.Root.Key, "a/b/c", "a/c")
+	require.NoError(t, err)
+	// check source files no longer exists
+	li, err := client.ListPath(ctx, buck.Root.Key, "a/b/c")
+	require.Error(t, err)
+
+	// check source parent remains untouched
+	li, err = client.ListPath(ctx, buck.Root.Key, "a/b")
+	require.NoError(t, err)
+	assert.True(t, li.Item.IsDir)
+	assert.Len(t, li.Item.Items, 0)
+
+	// move a dir into an existing path => a/b/c
+	_, err = client.MovePath(ctx, buck.Root.Key, "a/c", "a/b")
+	require.NoError(t, err)
+	// check source dir no longer exists
+	li, err = client.ListPath(ctx, buck.Root.Key, "a/c")
+	require.Error(t, err)
+
+	// check source parent contains all the right children
+	li, err = client.ListPath(ctx, buck.Root.Key, "a")
+	require.NoError(t, err)
+	assert.True(t, li.Item.IsDir)
+	assert.Len(t, li.Item.Items, 1)
+
+	// check source parent contains all the right children
+	li, err = client.ListPath(ctx, buck.Root.Key, "a/b/c")
+	require.NoError(t, err)
+	assert.True(t, li.Item.IsDir)
+	assert.Len(t, li.Item.Items, 2)
+
+	// move and rename file => a/b/c/file2.jpg + a/b/file3.jpg
+	_, err = client.MovePath(ctx, buck.Root.Key, "a/b/c/file1.jpg", "a/b/file3.jpg")
+	require.NoError(t, err)
+
+	li, err = client.ListPath(ctx, buck.Root.Key, "a/b/file3.jpg")
+	require.NoError(t, err)
+	assert.False(t, li.Item.IsDir)
+	assert.Equal(t, li.Item.Name, "file3.jpg")
+
+	li, err = client.ListPath(ctx, buck.Root.Key, "root.jpg")
+	require.NoError(t, err)
+
+	// move to root => /c
+	_, err = client.MovePath(ctx, buck.Root.Key, "a/b/c", "")
+	require.NoError(t, err)
+
+	li, err = client.ListPath(ctx, buck.Root.Key, "")
+	require.NoError(t, err)
+	assert.True(t, li.Item.IsDir)
+	assert.Len(t, li.Item.Items, 4)
+
+	// move root should fail
+	_, err = client.MovePath(ctx, buck.Root.Key, "", "a")
+	require.Error(t, err, "source is root directory")
+
+	li, err = client.ListPath(ctx, buck.Root.Key, "a")
+	require.NoError(t, err)
+	assert.True(t, li.Item.IsDir)
+	assert.Len(t, li.Item.Items, 1)
+
+	li, err = client.ListPath(ctx, buck.Root.Key, "")
+	require.NoError(t, err)
+	assert.True(t, li.Item.IsDir)
+	assert.Len(t, li.Item.Items, 4)
+
+	// move non existant should fail
+	_, err = client.MovePath(ctx, buck.Root.Key, "x", "a")
+	if private {
```

Comment:
unfortunate difference. i could change the errors to match in the first part of the string

Reply:
Ah yeah. We could catch the IPFS "no link named" error and throw "could not resolve path" instead. Maybe worth a ticket in general, "Make public and private bucket errors consistent" or something.

Reply:
k, will do. maybe we could implement a more detailed error type starting in here, that could return error types etc. it's a bit tricky to parse the ":" delimited stuff nicely. will move to a ticket though